classes/class-object-sync-sf-salesforce.php

```diff
@@ -1308,4 +1323,45 @@ private function cache_expiration() {
 		return $cache_expiration;
 	}
 
+	/*
+	* Get challenge code and verifier.
+	* If no cached challenge data, generate a new verifier and challenge code and cache them.
+	*/
+	protected function get_challenge() {
+		$data = get_transient( $this->option_prefix . 'challenge', '' );
+
+		if ( $data ) {
+			return $data;
+		}
```

Comment:
I would maybe code this more defensively, to make sure that `$data` is the expected shape.
```suggestion
		if ( is_array( $data ) && isset( $data['verifier], $data['code'] ) ) {
			return $data;
		}
```